lib/routers/establishment/projects.js

```diff
@@ -251,6 +295,19 @@ router.put('/:projectId/update-licence-holder',
   submit('update')
 );
 
+router.put('/:projectId/transfer-draft',
+  permissions('project.transfer'),
+  whitelist('targetEstablishmentId'),
+  (req, res, next) => {
+    res.response = req.project; // needed for fetchOpenTasks
+    next();
+  },
```

Comment:
This is a bit of a massive hack and is liable to end up causing the API to respond with dodgy data.

Could we modify the `fetchOpenTasks` middleware to accept a `getId` function as a parameter? This would have the side benefit of allowing the removal of the very similar hack at https://github.com/UKHomeOffice/asl-public-api/blob/master/lib/routers/establishment/project-versions.js#L125

Reply:
i.e. we'd replace this with

`fetchOpenTasks(req => req.project.id)`

And the other one with

`fetchOpenTasks(req => req.version.projectid)`